skyportal/tests/api/test_sources.py

```diff
@@ -1556,3 +1556,174 @@ def test_sources_hidden_photometry_not_leaked(
     assert data["data"]["id"] == obj_id
     assert len(public_source.photometry) - 1 == len(data["data"]["photometry"])
     assert photometry_id not in map(lambda x: x["id"], data["data"]["photometry"])
+
+
+def test_filter_sources_by_created_at(upload_data_token, view_only_token, public_group):
+    obj_id1 = str(uuid.uuid4())
+    obj_id2 = str(uuid.uuid4())
+
+    time_before_both = datetime.now(timezone.utc)
+
+    # Upload two new sources
+    status, data = api(
+        "POST",
+        "sources",
+        data={
+            "id": obj_id1,
+            "ra": 234.22,
+            "dec": -22.33,
+            "group_ids": [public_group.id],
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert data["data"]["id"] == obj_id1
+
+    partition_time = datetime.now(timezone.utc)
+
+    status, data = api(
+        "POST",
+        "sources",
+        data={
+            "id": obj_id2,
+            "ra": 234.22,
+            "dec": -22.33,
+            "group_ids": [public_group.id],
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert data["data"]["id"] == obj_id2
+
+    time_after_both = datetime.now(timezone.utc)
+
+    # Filter for obj 2 only
+    status, data = api(
+        "GET",
+        "sources",
+        params={
+            "createdOrModifiedAfter": str(partition_time),
+            "group_ids": f"{public_group.id}",
+        },
+        token=view_only_token,
+    )
+    assert status == 200
+    assert len(data["data"]["sources"]) == 1
+    assert data["data"]["sources"][0]["id"] == obj_id2
+
+    # Fetch both
+    status, data = api(
+        "GET",
+        "sources",
+        params={
+            "createdOrModifiedAfter": str(time_before_both),
+            "group_ids": f"{public_group.id}",
+        },
+        token=view_only_token,
+    )
+    assert status == 200
+    assert len(data["data"]["sources"]) == 2
+
+    # Filter both out
+    status, data = api(
+        "GET",
+        "sources",
+        params={
+            "createdOrModifiedAfter": str(time_after_both),
+            "group_ids": f"{public_group.id}",
+        },
+        token=view_only_token,
+    )
+    assert status == 200
+    assert len(data["data"]["sources"]) == 0
+
+
+def test_filter_sources_by_modified(upload_data_token, view_only_token, public_group):
+    obj_id1 = str(uuid.uuid4())
+    obj_id2 = str(uuid.uuid4())
+
+    time_before_both = datetime.now(timezone.utc)
```

Comment:
Maybe it would be quicker and shorter if we just use the fixture sources and modify them?

Reply:
But I want to be sure we're filtering on created_at, too. Suggest we leave as-is.